plugins/woocommerce/tests/php/src/Admin/ProductBlockEditor/ProductTemplates/CustomProductFormTemplateTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Automattic\WooCommerce\Tests\Admin\ProductBlockEditor\ProductTemplates;
+
+use WC_Unit_Test_Case;
+
+/**
+ * Tests for the CustomBlockTemplate class.
+ */
+class CustomProductFormTemplateTest extends WC_Unit_Test_Case {
+	/**
+	 * Test getting the template ID.
+	 */
+	public function test_get_id() {
+		$template = new CustomProductFormTemplate();
+		$this->assertEquals( $template->get_id(), 'custom-product' );
+	}
+
+	/**
+	 * Test get group block by id.
+	 */
+	public function test_get_group_by_id() {
+		$template = new CustomProductFormTemplate();
+		$block = $template->get_group_by_id('general' );
+		$formatted_template = $block->get_formatted_template();
+		$this->assertEquals( $formatted_template[1]['title'], 'General' );
+	}
+
+	/**
+	 * Test throw error get group block with section id.
+	 */
+	public function test_get_group_by_id_with_section_id() {
+		$template = new CustomProductFormTemplate();
+
+		$this->expectException( \UnexpectedValueException::class );
+		$template->get_group_by_id('basic-details' );
+	}
+
+	/**
+	 * Test get section block by id.
+	 */
+	public function test_get_section_by_id() {
+		$template = new CustomProductFormTemplate();
+		$block = $template->get_section_by_id('product-pricing-section' );
+		$formatted_template = $block->get_formatted_template();
+		$this->assertEquals( $formatted_template[1]['title'], 'Pricing' );
```

Comment:
Once `SectionInterface` extends `BlockInterface` (which is should, since a group is a block), you can do the following:

```php
$this->assertEquals( $block->get_attributes()['title'], 'Pricing' );
```